src/main/java/in/org/projecteka/hiu/clients/HealthInformationClient.java

```diff
@@ -0,0 +1,34 @@
+package in.org.projecteka.hiu.clients;
+
+import in.org.projecteka.hiu.common.CentralRegistry;
+import lombok.AllArgsConstructor;
+import org.apache.log4j.Logger;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import static java.util.function.Predicate.not;
+
+@AllArgsConstructor
+public class HealthInformationClient {
+    private final WebClient.Builder builder;
+    private final CentralRegistry centralRegistry;
+    private final Logger logger = Logger.getLogger(CentralRegistryClient.class);
+
+    public Mono<HealthInformation> getHealthInformationFor(String url) {
+        return centralRegistry.token()
+                .flatMap(token -> builder.build()
```

Comment:
We should not get token from central registry for pulling data from link as it's already secure with temporary token, I believe. Because it could be from anywhere we could pull data S3/dropbox/anything.

Reply:
oh yes. I will remove it.